components/admin_console/jobs/table.tsx

```diff
@@ -1,82 +1,47 @@
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
 
-import {FormattedDate, FormattedMessage, FormattedTime, injectIntl} from 'react-intl';
+import {FormattedDate, FormattedMessage, FormattedTime, injectIntl, IntlShape} from 'react-intl';
 
 import {Client4} from 'mattermost-redux/client';
+import {getJobsByType, createJob, cancelJob} from 'mattermost-redux/actions/jobs';
+import {Job, JobStatus, JobType} from 'mattermost-redux/types/jobs';
 
 import {JobStatuses, exportFormats, JobTypes} from 'utils/constants';
-import {intlShape} from 'utils/react_intl';
 import * as Utils from 'utils/utils.jsx';
 import './table.scss';
 
-class JobTable extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * Used for formatting dates
-         */
-        intl: intlShape.isRequired,
-
-        /**
-         * Array of jobs
-         */
-        jobs: PropTypes.arrayOf(PropTypes.object).isRequired,
-
-        /**
-         * Function called when displaying extra text.
-         */
-        getExtraInfoText: PropTypes.func,
-
-        /**
-         * Grey buttons out when disabled
-         */
-        disabled: PropTypes.bool,
-
-        /**
-         * Help text under the create job button
-         */
-        createJobHelpText: PropTypes.element.isRequired,
-
-        /**
-         * Button text to create a new job
-         */
-        createJobButtonText: PropTypes.element.isRequired,
-
-        /**
-         * The type of jobs to include in this table.
-         */
-        jobType: PropTypes.string.isRequired,
-
-        /**
-         * A variable set in config.json to determine if results can be downloaded or not.
-         * Note that there is NO front-end associated with this setting due to security.
-         * Only the person with access to the config.json file can enable this option.
-         */
-        downloadExportResults: PropTypes.bool,
-
-        actions: PropTypes.shape({
-            getJobsByType: PropTypes.func.isRequired,
-            cancelJob: PropTypes.func.isRequired,
-            createJob: PropTypes.func.isRequired,
-        }).isRequired,
-
-        /**
-         * Allows for custom styles on the job table component
-         */
-        className: PropTypes.string,
-
-        /**
-         * Hide the job creation button. This is useful if you want to place the button elsewhere on your page or hide it.
-         */
-        hideJobCreateButton: PropTypes.bool,
+type ArgumentTypes<F extends (...args: any) => any> = F extends (...args: infer A) => any ? A : never;
+type FuncFromArgAndReturn<A extends (...args: any) => any> = (...args: ArgumentTypes<A>) => any;
+
+export type Props = {
+    jobs?: Job[];
+    getExtraInfoText?: (job: Job) => React.ReactElement;
+    disabled?: boolean;
+    createJobHelpText: React.ReactElement;
+    createJobButtonText: React.ReactElement;
+    jobType: JobType;
+    downloadExportResults?: boolean;
+    actions?: {
```

Comment:
```suggestion
    actions: {
```

Since actions are always defined, we can remove the `?` optional question mark.

Reply:
Sure